Make the app cross-platform by using Path.Join

amir734jj opened the pull request:
- Fixed login bug by using OR instead of AND (probably a logic bug)
- Used `await using` instead of `using` which is a new C# feature
- Added support for Linux and Mac by using cross-platform `Path.Join` as oppose to string interpolation

amir734jj committed:
Make the app cross-platform by using Path.Join

TahaNKhan commented on Deployr.Web/Logic/DeployLogic.cs:
```diff
@@ -58,39 +58,39 @@ public async Task<BasicResponse> UploadPackageAsync(int deploymentId, IFormFile
 		public async Task<BasicResponse> CreateDeployment(CreateDeploymentRequest metadata)
 		{
-			using var dataContext = await _dataContextFactory.Construct();
+			await using var dataContext = await _dataContextFactory.Construct();
```
Ooo fancy

TahaNKhan commented on Deployr.Web/Logic/DeployLogic.cs:
```diff
@@ -108,7 +108,7 @@ private DirectoryInfo ValidateAndCleanDirectory(string deployLocation, bool forc
 
 			// Delete old artifacts
-			if (directoryExists && forceDeploy)
+			if (directoryExists || forceDeploy)
```
We should make sure the directory exists before we delete it, right?

TahaNKhan commented on Deployr.Web/Logic/DeployLogic.cs:
```diff
@@ -117,20 +117,24 @@ private DirectoryInfo ValidateAndCleanDirectory(string deployLocation, bool forc
 		private static string DetermineDefaultDeployLocation()
 		{
 			const string windowsDefaultDeployLocation = "C:\\CustomApps";
+			const string linuxDefaultDeployLocation = "~/CustomApps";
```
Would /use/local/bin be appropriate here? I’m not too familiar with Linux directory structure anymore.

amir734jj committed:
Updated the code according to the suggestions

amir734jj commented:
Updated the code. Removed `forceDeploy` because it's redundant in the second if statement. Rider will show it's reducent because it will be always true.
